sql: remove some leftover tenant-specific stuff

yuzefovich opened the pull request:
Informs: #116445.

Epic: None

Release note: None

yuzefovich committed:
sql: remove leftover system-tenant specific version code

This commit removes usage of `ForSystemTenant()` bool value which was
used when updating the version cluster setting, and it appears to only
be necessary around 20.2-21.1 migration. We shouldn't need to have
a difference in behavior between system and secondary tenants anymore.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/116457)
<!-- Reviewable:end -->

yuzefovich committed:
sql: remove unused planFlagTenant

This was introduced in 5f05e15b8f1037de8344efbb6563bd9af321e844, but
then the refactor in e94dc792c395383c3fa31a1735c520029573267b removed
its only usage.

Release note: None

yuzefovich committed:
sql: fix up a comment

Release note: None

yuzefovich commented:
TFTR!

bors r+
